packages/core/src/electron-browser/window/electron-window-service.ts

```diff
@@ -36,11 +38,24 @@ export class ElectronWindowService extends DefaultWindowService {
     @inject(ElectronMainWindowService)
     protected readonly delegate: ElectronMainWindowService;
 
+    @inject(WindowPreferences)
+    protected readonly windowPreferences: WindowPreferences;
+
     openNewWindow(url: string, { external }: NewWindowOptions = {}): undefined {
         this.delegate.openNewWindow(url, { external });
         return undefined;
     }
 
+    @postConstruct()
+    protected init(): void {
+        // if window.zoomLevel is not default (0) on startup, `Preferences` will fire event
+        this.windowPreferences.onPreferenceChanged((e: PreferenceChangeEvent<WindowConfiguration>) => {
```

Comment:
You can omit the signature:

```suggestion
        this.windowPreferences.onPreferenceChanged(e => {
```